src/components/settings/LocationSettings.js

```diff
@@ -0,0 +1,163 @@
+import * as Location from 'expo-location';
+import React, { useEffect, useState } from 'react';
+import { Alert, View } from 'react-native';
+import Collapsible from 'react-native-collapsible';
+
+import { colors, texts } from '../../config';
+import { useLocationSettings } from '../../hooks';
+import { WebViewMap } from '../map';
+import { Button } from '../Button';
+import { LoadingSpinner } from '../LoadingSpinner';
+import { RegularText } from '../Text';
+import { ToggleListItem } from '../ToggleListItem';
+import { Wrapper, WrapperWithOrientation } from '../Wrapper';
+import { location, locationIconAnchor } from '../../icons';
+import { latLngToLocationObject } from '../../helpers';
+
+const baseLocationMarker = {
+  icon: location(colors.primary),
+  iconAnchor: locationIconAnchor
+};
+
+const useSystemPermission = () => {
+  const [systemPermission, setSystemPermission] = useState();
+
+  useEffect(() => {
+    (async () => setSystemPermission(await Location.getForegroundPermissionsAsync()))();
+  }, []);
+
+  return systemPermission;
+};
+
+const getLocationMarker = (locationObject) => ({
+  ...baseLocationMarker,
+  id: 'alternativePosition',
+  position: {
+    lat: locationObject.coords.latitude,
+    lng: locationObject.coords.longitude
+  }
+});
+
+export const LocationSettings = () => {
+  const { locationSettings, setAndSyncLocationSettings } = useLocationSettings();
+  const systemPermission = useSystemPermission();
+
+  const [showMap, setShowMap] = useState(false);
+  const [selectedPosition, setSelectedPosition] = useState();
+
+  if (!systemPermission) {
+    return <LoadingSpinner loading />;
```

Comment:
do we need to wrap it somehow to position it on the screen? is there a good way to test this case within Expo app?

Reply:
`LoadingSpinner` uses the `LoadingContainer` internally. Do you think we should another method?

Reply:
a nice did not remember that